File: app/src/main/java/com/bigscreen/mangindo/stored/StoredDataModule.java
package com.bigscreen.mangindo.stored;


import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class StoredDataModule {

    @Provides
    @Singleton
    @SuppressWarnings("unused")
    public PreferenceService providesPreferenceService(Context context) {
        return new PreferenceService(context);
    }

    @Provides
    @Singleton
    @SuppressWarnings("unused")
    public StoredDataService providesStoredDataService(PreferenceService preferenceService) {
        return new StoredDataService(preferenceService);
    }
}

File: app/src/main/java/com/bigscreen/mangindo/stored/StoredDataModule.kt
package com.bigscreen.mangindo.stored

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class StoredDataModule {

    @Provides
    @Singleton
    fun providesPreferenceService(context: Context): PreferenceService {
        return PreferenceService(context)
    }

    @Provides
    @Singleton
    fun providesStoredDataService(preferenceService: PreferenceService): StoredDataService {
        return StoredDataService(preferenceService)
    }
}
